Apps/Src/Amps.c

```diff
@@ -31,18 +47,22 @@ ErrorStatus Amps_UpdateMeasurements(void) {
  * @return SAFE or DANGER
  */
 SafetyStatus Amps_CheckStatus(bool chargingOnly) {
-	if(chargingOnly) {
-		// Only allow charging status
-		if((latestMeasureMilliAmps <= 0) 
-		 &&(latestMeasureMilliAmps > MAX_CHARGING_CURRENT)) {
-			return SAFE;
-		}
-	} else {
-		// Constrain between the max charging allowed (negative) and the max draw (postive)
-		if((latestMeasureMilliAmps > MAX_CHARGING_CURRENT) 
-		 &&(latestMeasureMilliAmps < MAX_CURRENT_LIMIT)) {
-			 return SAFE;
-		}
+	OSMutexPend(&AmperesData_Mutex, 0, OS_OPT_PEND_BLOCKING, &ticks, &err);
+	assertOSError(err);
+	if((latestMeasureMilliAmps > MAX_CHARGING_CURRENT)&&(latestMeasureMilliAmps < MAX_CURRENT_LIMIT)&&(!chargingOnly)){
+		OSMutexPost(&AmperesData_Mutex, OS_OPT_POST_NONE, &err);
+		assertOSError(err);
+		return SAFE;
+	}
+	else if((latestMeasureMilliAmps <= 0)&&(latestMeasureMilliAmps > MAX_CHARGING_CURRENT)&&chargingOnly){
+		OSMutexPost(&AmperesData_Mutex, OS_OPT_POST_NONE, &err);
+		assertOSError(err);
+		return SAFE;
+	}
+	else{
+		OSMutexPost(&AmperesData_Mutex, OS_OPT_POST_NONE, &err);
+		assertOSError(err);
+		return DANGER;
 	}
 
 	return DANGER;
```

Comment:
Could simplify this all quite a bit by doing something like
```suggestion
	SafetyStatus stat = DANGER;
	OSMutexPend(&AmperesData_Mutex, 0, OS_OPT_PEND_BLOCKING, &ticks, &err);
	assertOSError(err);
	if((latestMeasureMilliAmps > MAX_CHARGING_CURRENT)&&(latestMeasureMilliAmps < MAX_CURRENT_LIMIT)&&(!chargingOnly)){
		stat = SAFE;
	}
	else if((latestMeasureMilliAmps <= 0)&&(latestMeasureMilliAmps > MAX_CHARGING_CURRENT)&&chargingOnly){
		stat = SAFE;
	}
	else{
		stat = DANGER;
	}

	OsMutexPost(&AmperesData_Mutex, OS_OPT_POST_NONE, &err);
	assertOSError(err);
	return stat;
```